File: python/p010b.py
#!/usr/bin/python
# run time real	0m1.343s

from math import sqrt
from math import floor
from funfunctions import primetest

maxvalue = 2000000

sievebound = (maxvalue - 1)/2
maxindex = int((floor(sqrt(maxvalue))-1)/2)

sieve = [True]*sievebound

for index in range(1, maxindex):
	if sieve[index]:
		for jndex in range((2*index*(index+1)),sievebound,(2*index+1)):
			sieve[jndex] = False

aggregator = 2

for index in range(1, sievebound):
	if sieve[index]: 
		aggregator += (2*index+1)

	
print aggregator

File: ruby/p010b.rb

CAP = 2_000_000

sieve_bound = (CAP - 1)/2
max_idx = ((Math.sqrt(CAP)-1).floor/2).to_i

sieve = Array.new(sieve_bound) { true }

(1..max_idx).each do |idx|
  if sieve[idx]
    ((2 * idx * (idx + 1))..sieve_bound).step(2 * idx + 1) do |j|
      sieve[j] = false
    end
  end
end

aggregator = 2

(1..sieve_bound).each do |i|
  aggregator += (2 * i + 1) if sieve[i]
end

puts aggregator
